test/src/unit-vfs.cc

```diff
@@ -622,8 +622,8 @@ TEST_CASE("VFS: test ls_with_sizes", "[vfs][ls-with-sizes]") {
   require_tiledb_ok(vfs_ls.remove_dir(URI(path)));
 }
 
-// Currently only S3 is supported for VFS::ls_recursive.
-using TestBackends = std::tuple<S3Test>;
+// Currently only S3 and Azure is supported for VFS::ls_recursive.
+using TestBackends = std::tuple<S3Test, AzureTest>;
 TEMPLATE_LIST_TEST_CASE(
```

Comment:
For tests using custom filters we could update these tests

https://github.com/TileDB-Inc/TileDB/blob/dev/test/src/unit-cppapi-vfs.cc#L510-L649

It looks like only one of them is using TEMPLATE_LIST_TEST_CASE at the moment but if it's not too much trouble I think all three of them could be updated to do this for local, s3, and azure.

Reply:
Similar tests for C APIs here that could be updated

https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/api/c_api/vfs/test/unit_capi_ls_recursive.cc

Reply:
The local filesystem seems to fail. Investigating…